### PR TITLE
apps: Fix make distclean for CONFIG_BUILD_KERNEL=y

### PR DESCRIPTION
**Pull request by masayuki2009**
## Summary

- I noticed that some files (e.g. xxx.tar.gz) which are generated
   by 'make context' are not deleted when we build the kernel only
    then we do 'make distclean' because .depend is not generated.
- This commit fixes this issue by adding the .context file and
    also adding directories that contain the file to be cleaned.

## Impact

- None

## Testing

- Tested with sabre-6quad:netknsh (not merged yet)

### CODE REVIEW

File: Directory.mk [hunk 1]
```diff
@@ -26,6 +26,7 @@ SUBDIRS       := $(dir $(wildcard *$(DELIM)Makefile))
 CONFIGSUBDIRS := $(filter-out $(dir $(wildcard *$(DELIM)Kconfig)),$(SUBDIRS))
 CLEANSUBDIRS  += $(dir $(wildcard *$(DELIM).depend))
 CLEANSUBDIRS  += $(dir $(wildcard *$(DELIM).kconfig))
+CLEANSUBDIRS  += $(dir $(wildcard *$(DELIM)*.tar.*))
```
Comment (Ouss4) [inline Directory.mk:29]: Not sure why is this required.  I see the discussion here https://github.com/apache/incubator-nuttx/pull/6586

`.depend` is created in the argtable3 directory and thus `distclean` should execute there and delete the downloaded files.

Comment (masayuki2009) [reply Directory.mk:29]: @Ouss4 
'.depend' is NOT created when building the kernel (CONFIG_BUILD_KERNEL=y) only and will be created when we do 'make export'

Comment (Ouss4) [reply Directory.mk:29]: Ah I see.  Thanks for the explanation.

Comment (xiaoxiang781216) [reply Directory.mk:29]: But this fix isn't extensible since not all downloaded package contain *.tar.*

Comment (masayuki2009) [reply Directory.mk:29]: >But this fix isn't extensible since not all downloaded package contain .tar.

@xiaoxiang781216 
How about creating the .context file when downloading a file such as .tar.gz or generating a new file such as config.h
Then .context will be checked when executing 'make distclean'

Comment (xiaoxiang781216) [reply Directory.mk:29]: Yes, I think it's better to generate .context in kernel mode just like other mode.

**Comment by masayuki2009** [comment]
@xiaoxiang781216 
Please review this PR again.

**Comment by xiaoxiang781216** [comment]
@masayuki2009 how about we invoke "make distclean) in all subfolder like https://github.com/apache/incubator-nuttx-apps/pull/1217?

**Comment by masayuki2009** [comment]
> @masayuki2009 how about we invoke "make distclean in all subfolder like #1217?

I think the PR will take a longer time because the PR always traverse all app directories which contain Makefile even though the nuttx configuration does not use them.

**Comment by xiaoxiang781216** [comment]
Yes, but it could ensure all clean/distclean target get executed. On the other hand, the method used to detect whether the directories is part of the build is fragile.

**Comment by masayuki2009** [comment]
Let me close this PR, since https://github.com/apache/incubator-nuttx/pull/6719 fixes the original issue.
